pwnlib/util/fiddling.py

```diff
@@ -370,7 +370,7 @@ def xor_pair(data, avoid = b'\x00\n'):
     if isinstance(data, six.integer_types):
         data = packing.pack(data)
 
-    if not isinstance(avoid, bytes):
+    if not isinstance(avoid, (bytes, bytearray)):
         avoid = avoid.encode('utf-8')
 
     avoid = bytearray(avoid)
```

Comment:
If `avoid` is already a `bytearray` this will fail on the `encode`.  We should probably add a special case here too?

Reply:
Yeah, that's why we're not encoding if `avoid` is either `bytes` or `bytearray`.